File: commands/wheel.py
#!/usr/bin/env python3
import time
import discord
import random
import modules.megacoin as megacoin


async def wheel(ctx, wager):

    balance = await megacoin.balance(ctx.user)

    if wager < 0:
        embed = discord.Embed(
            color=0x9366CD, title="☸️  Spin the Wheel", description="🤡  Nice try."
        )
        await ctx.respond(embed=embed)
        return
    elif balance == 0:
        embed = discord.Embed(
            color=0x9366CD,
            title="☸️  Spin the Wheel",
            description="You have 0 <:MEGACOIN:1090620048621707324>",
        )
        await ctx.respond(embed=embed)
        return
    elif balance < wager:
        embed = discord.Embed(
            color=0x9366CD,
            title="☸️  Spin the Wheel",
            description=f"You don't have {str(wager)} <:MEGACOIN:1090620048621707324>",
        )
        await ctx.respond(embed=embed)
        return

    spin = random.choice([0, 0.5, 0.5, 1, 1, 1.5, 1.5, 2])
    diff = random.choice([1, 2])
    win = int(wager * spin)

    url = f"https://raw.githubusercontent.com/NicPWNs/MEGABOT/main/images/{str(spin)}-{str(diff)}.gif"
    embed = discord.Embed(
        color=0x9366CD, title="☸️  Spin the Wheel", description="Spinning..."
    ).set_thumbnail(url=url)
    interaction = await ctx.respond(embed=embed)

    time.sleep(5)
    await megacoin.subtract(ctx.user, wager)
    await megacoin.add(ctx.user, win)

    description = f"<@{ctx.user.id}> wagered {str(wager)} <:MEGACOIN:1090620048621707324> and **won {str(win)}** <:MEGACOIN:1090620048621707324>!"
    url = f"https://raw.githubusercontent.com/NicPWNs/MEGABOT/main/images/{str(spin)}-{str(diff)}.jpg"
    embed = (
        discord.Embed(
            color=0x9366CD, title="☸️  Spin the Wheel", description=description
        )
        .set_thumbnail(url=url)
        .set_footer(text=f"Their balance is now {await megacoin.balance(ctx.user)}")
    )

    await interaction.edit_original_response(embed=embed)

File: commands/wheel.ts
import {
  DiscordInteraction,
  InteractionResponse,
  MEGACOIN_EMOJI,
} from "../types/discord";
import { embedResponse } from "../utils/discord";
import { getOptionValue, getUser } from "../handlers/commands";
import { getBalance, addCoins, subtractCoins } from "../services/megacoin";

// Wheel multipliers matching Python: [0, 0.5, 0.5, 1, 1, 1.5, 1.5, 2]
const WHEEL_OPTIONS = [0, 0.5, 0.5, 1, 1, 1.5, 1.5, 2];

function spinWheel(): { multiplier: number; variant: number } {
  const multiplier =
    WHEEL_OPTIONS[Math.floor(Math.random() * WHEEL_OPTIONS.length)];
  const variant = Math.random() < 0.5 ? 1 : 2; // Random 1 or 2 for image variant
  return { multiplier, variant };
}

export async function handleWheel(
  interaction: DiscordInteraction,
): Promise<InteractionResponse> {
  const wager = getOptionValue<number>(interaction, "wager")!;
  const user = getUser(interaction);

  // Validation (matches Python)
  if (wager < 0) {
    return embedResponse({
      title: "☸️ Spin the Wheel",
      description: "🤡 Nice try.",
      color: 0x9366cd,
    });
  }

  const balance = await getBalance(user.id);

  if (balance === 0) {
    return embedResponse({
      title: "☸️ Spin the Wheel",
      description: `You have 0 ${MEGACOIN_EMOJI}`,
      color: 0x9366cd,
    });
  }

  if (balance < wager) {
    return embedResponse({
      title: "☸️ Spin the Wheel",
      description: `You don't have ${wager} ${MEGACOIN_EMOJI}`,
      color: 0x9366cd,
    });
  }

  // Spin the wheel
  const { multiplier, variant } = spinWheel();
  const winnings = Math.floor(wager * multiplier);

  // Apply the bet
  await subtractCoins(user.id, user.username, wager);
  if (winnings > 0) {
    await addCoins(user.id, user.username, winnings);
  }
  const newBalance = await getBalance(user.id);

  // Use wheel result image from GitHub (matches Python)
  const imageUrl = `https://raw.githubusercontent.com/NicPWNs/MEGABOT/main/images/${multiplier}-${variant}.jpg`;

  return embedResponse({
    title: "☸️ Spin the Wheel",
    description: `<@${user.id}> wagered ${wager} ${MEGACOIN_EMOJI} and **won ${winnings}** ${MEGACOIN_EMOJI}!`,
    color: 0x9366cd,
    thumbnail: { url: imageUrl },
    footer: { text: `Their balance is now ${newBalance}` },
  });
}
